genshin/utility/auth.py

```diff
@@ -116,3 +118,11 @@ def encrypt_geetest_credentials(text: str, region: Region = Region.OVERSEAS) ->
 def get_aigis_header(session_id: str, mmt_data: typing.Dict[str, typing.Any]) -> str:
     """Get aigis header."""
     return f"{session_id};{base64.b64encode(json.dumps(mmt_data).encode()).decode()}"
+
+
+def generate_sign(data: typing.Dict[str, typing.Any], key: str) -> str:
+    """Generate a sign for the given `data` and `app_key`."""
+    string = ""
+    for k in sorted(data.keys()):
```

Comment:
Is the sorting required here?

Reply:
I haven't checked if request succeed without sorting, but it would be better to maintain consistency with the devs' implementation

Reply:
Alright